presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1808,53 +1809,151 @@ private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope so
 
             for (SelectItem item : node.getSelect().getSelectItems()) {
                 if (item instanceof AllColumns) {
-                    // expand * and T.*
-                    Optional<QualifiedName> starPrefix = ((AllColumns) item).getPrefix();
+                    analyzeSelectAllColumns((AllColumns) item, node, scope, outputExpressionBuilder);
+                }
+                else if (item instanceof SingleColumn) {
+                    analyzeSelectSingleColumn((SingleColumn) item, node, scope, outputExpressionBuilder);
+                }
+                else {
+                    throw new IllegalArgumentException("Unsupported SelectItem type: " + item.getClass().getName());
+                }
+            }
+
+            ImmutableList<Expression> result = outputExpressionBuilder.build();
+            analysis.setOutputExpressions(node, result);
 
+            return result;
+        }
+
+        private void analyzeSelectAllColumns(AllColumns allColumns, QuerySpecification node, Scope scope, ImmutableList.Builder<Expression> outputExpressionBuilder)
+        {
+            // expand * and expression.*
+            if (allColumns.getTarget().isPresent()) {
+                // analyze AllColumns with target expression (expression.*)
+                Expression expression = allColumns.getTarget().get();
+
+                // try resolve target expression as table reference or column reference
+                QualifiedName prefix = null;
+                if (expression instanceof DereferenceExpression) {
+                    prefix = DereferenceExpression.getQualifiedName((DereferenceExpression) expression);
+                }
+                else if (expression instanceof Identifier) {
+                    prefix = QualifiedName.of(ImmutableList.of((Identifier) expression));
+                }
+                if (prefix != null) {
+                    // try match prefix to a table
                     RelationType relationType = scope.getRelationType();
-                    List<Field> fields = relationType.resolveFieldsWithPrefix(starPrefix);
-                    if (fields.isEmpty()) {
-                        if (starPrefix.isPresent()) {
-                            throw new SemanticException(MISSING_TABLE, item, "Table '%s' not found", starPrefix.get());
-                        }
-                        if (!node.getFrom().isPresent()) {
-                            throw new SemanticException(WILDCARD_WITHOUT_FROM, item, "SELECT * not allowed in queries without FROM clause");
+                    List<Field> fields = relationType.resolveFieldsWithPrefix(Optional.of(prefix));
+
+                    if (!fields.isEmpty()) {
+                        // successfully dereferenced a table
+                        if (!allColumns.getAliases().isEmpty()) {
+                            throw new SemanticException(INVALID_PARAMETER_USAGE, node.getSelect(), "SELECT t.* FROM table does not allow column aliases");
```

Comment:
`INVALID_PARAMETER_USAGE` doesn't seem like the right error code for this. We probably need a new one to handle this scenario.